### PR TITLE
Expand command line tool documentation

### PR DESCRIPTION
**Pull request by melissalinkert**
See http://trac.openmicroscopy.org/ome/ticket/10329 and https://trello.com/c/pSlJ3apG/94-review-existing-bf-dev-docs.  All command line tools should now have at least a brief description.

### CODE REVIEW

File: docs/sphinx/developers/useful-scripts.txt [hunk 1]
```diff
@@ -0,0 +1,41 @@
+Scripts for performing development tasks
+========================================
+
+The :code:`tools` directory contains several scripts which are useful
```
Comment (mtbc) [inline docs/sphinx/developers/useful-scripts.txt]: `:file:` instead?

File: docs/sphinx/developers/useful-scripts.txt [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+This updates the Maven POM version numbers for all pom.xml files that
+set `groupId` to `ome`.  The script takes a single argument, which is the new
+version.  For example, to update the POM versions prior to release::
+
+  ./tools/bump_maven_version.py 5.1.0
+
+and to switch back to snapshot versions immediately after release::
+
+  ./tools/bump_maven_version.py 5.1.1-SNAPSHOT
```
Comment (mtbc) [inline docs/sphinx/developers/useful-scripts.txt]: (Outwith this PR: It's interesting that the usual maven release plugin stuff like update-versions isn't sufficing for us.)

File: docs/sphinx/users/comlinetools/conversion.txt [hunk 3]
```diff
@@ -67,3 +67,16 @@ example, to use LZW compression in a TIFF file:
 
   bfconvert -compression LZW /path/to/input output-lzw.tiff
 
+If the specified output file already exists, bfconvert will prompt to
```
Comment (mtbc) [inline docs/sphinx/users/comlinetools/conversion.txt]: I don't know if we want `:command:` output for these, maybe not; just a thought.

File: docs/sphinx/developers/useful-scripts.txt (hunk 2 above)
Comment (sbesson) [inline docs/sphinx/developers/useful-scripts.txt]: I don't think the `maven` release plugin will be able to update the version in the UpgradeCheck.java class

File: docs/sphinx/users/comlinetools/index.txt [hunk 4]
```diff
@@ -55,6 +56,16 @@ Currently available tools include:
...
+    domainlist
+        Displays a list of imaging domains and the supported formats
+        associated with each domain.  See :doc:`domainlist` for more
+        information.
+
+    mkfake
+        Creates a "fake" high-content screen with configurable dimensions.
+        This is useful for testing how HCS metadata is handled, without
+        requiring real image data from an acquired screen.  See :doc:`mkfake`
+        for more information.
```
Comment (mtbc) [inline docs/sphinx/users/comlinetools/index.txt]: I'm guessing that the above list is largely in descending order of how much we expect a user to want to use the tool.

File: docs/sphinx/developers/useful-scripts.txt (hunk 2 above)
Comment (mtbc) [inline docs/sphinx/developers/useful-scripts.txt]: Ah, that might be it, and we can't use maven filtering to fix that if supporting ant too; thank you.

File: docs/sphinx/users/comlinetools/mkfake.txt [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+  mkfake -columns 12 twelve-column-plate.fake
+
+To change the number of fields per well:
+
+::
+
+  mkfake -fields 2 two-field-plate.fake
+
+It's often most useful to use the arguments together to create a realistic
```
Comment (mtbc) [inline docs/sphinx/users/comlinetools/mkfake.txt]: "It is"

**Comment by mtbc** [comment]
Generally helpful and nicely written. :+1:

File: docs/sphinx/users/comlinetools/index.txt (hunk 4 above)
Comment (melissalinkert) [inline docs/sphinx/users/comlinetools/index.txt]: Mostly they are in the order in which the pages were added, which I guess sort of happens to coincide with how generally useful each command is.  I can certainly reorder them if desired.

**Comment by melissalinkert** [comment]
Commit pushed for formatting/style fixes.

File: docs/sphinx/users/comlinetools/index.txt (hunk 4 above)
Comment (mtbc) [inline docs/sphinx/users/comlinetools/index.txt]: If they are accidentally in descending order of how much we expect a user to want to use the tool then that works for me. (Or lexicographical or whatever else.)

**Comment by ximenesuk** [comment]
Final changes look good, I don't know if this needs further review, @mtbc

File: docs/sphinx/users/comlinetools/display.txt [hunk 6]
```diff
@@ -59,7 +59,6 @@ coordinate (0, 0) is the upper-left corner of the image; ``x + width`` must be
 less than or equal to the image width and ``y + height`` must be less than or
 equal to the image height.
 
-
 By default, ``showinf`` will check for a new version of Bio-Formats.  This can
```
Comment (mtbc) [inline docs/sphinx/users/comlinetools/display.txt]: probably also want `:command:` markup in general for these other commands

**Comment by melissalinkert** [comment]
I think all of the places where ```:command:``` formatting was missing are fixed now.

**Comment by mtbc** [comment]
Latest changes look good too. :+1:

**Comment by melissalinkert** [comment]
--no-rebase

**Pull request merged**